octoprint_v8theme/static/js/v8theme.js

```diff
@@ -14,20 +164,269 @@ $(function() {
+                commandName = commandName.replace(/\(|\)|\|/g, "");
+                $(this).find("span").replaceWith("<span>Supress " + commandName + "</span>");
+                $(this).css('margin-bottom', '0');
+                $("ul.terminal-options").append(
+                    $('<li>').append(
+                        $('<a>').attr('href', '#').append(
+                            $(this)
+                        )))
+            });
+            $("#terminal-sendpanel .input-append input").addClass("form-control").attr("placeholder", "Enter a command...").appendTo(".terminal-textbox");
+            $("#terminal-sendpanel .input-append button").addClass("btn-default btn-gradient btn-block").appendTo(".terminal-submit");
+            $("#termin-filterpanel").parent().remove();
+            $(".terminal .pull-left, .terminal .pull-right").remove();
+
+            $(".temperature-height").click(function() {
+                $(".temperature-height").toggleClass("active");
+                $("#temperature-graph").animate({
+                    height: ($("#temperature-graph").height() == 250) ? 500 : 250
+                }, {
+                    duration: 250,
+                    step: function() {
+                        self.temperature.updatePlot();
+                    }
+                });
+                $("#tooltip_bar").animate({
+                    height: ($("#temperature-graph").height() == 250) ? 485 : 235
+                }, 250);
+            });
+            $("<div id='tooltip_bar'></div>").css({
+                position: "absolute",
+                "display": "none",
+                border: "1px solid #BFBFBF",
+                padding: "2px",
+                "background-color": "#000",
+                opacity: 0.04,
+                "height": "235px",
+                "width": "35px",
+                "top": "15px"
+            }).appendTo("#temperature_main .accordion-inner");
+            $("<div id='tooltip'><table><tbody></tbody></table></div>").appendTo("#temperature_main .accordion-inner");
+
+            var legends = $("#temperature-graph .legendLabel");
+            legends.each(function() {
+                $(this).css('width', $(this).width());
+            });
+
+            // Legend/tooltip for Flot chart
+            var updateLegendTimeout = null;
+            var latestPosition = null;
+
+            function updateLegend() {
+                updateLegendTimeout = null;
+                self.temperature.plot.unhighlight();
+                $("#tooltip_bar, #tooltip").css("display", "none");
+                var pos = latestPosition,
+                    axes = self.temperature.plot.getAxes();
+                if (pos.x < axes.xaxis.min || pos.x > axes.xaxis.max || pos.y < axes.yaxis.min || pos.y > axes.yaxis.max) return;
+                var dataset = self.temperature.plot.getData();
+                var halfDist = (dataset[0].data[1][0] - dataset[0].data[0][0]) / 2;
```

Comment:
This happens when you mouse over the temperature graph but `dataset` hasn't been defined yet. I know this happens upon reconnecting after OctoPrint restarts, etc. I'll just add a return if undefined statement.